### PR TITLE
SIS2: +Add transmute ice outflow open boundary condition

### PR DESCRIPTION
**Pull request by Hallberg-NOAA**
Added the new option to transmute sea ice and snow into sea water as a simple
form of outflow open boundary condition.  Also cleaned up some problems with
non-reproducing SIS_parameter_doc files.  There are new runtime options and
entries in the SIS_parameter_doc files, but all answers in the existing
MOM6-examples test suite are bitwise identical.  The commits in this PR include:

- NOAA-GFDL/SIS2@d5e7883 +Added TRANSMUTE_ICE option
- NOAA-GFDL/SIS2@acc8a6c +Corrected description of ICE_THICKNESS_INIT_CONFIG

### CODE REVIEW

**Comment by kshedstrom** [comment]
I approve this PR, but suggest this change:
```
diff --git a/src/SIS_slow_thermo.F90 b/src/SIS_slow_thermo.F90
index 1264b8f..48f1003 100644
--- a/src/SIS_slow_thermo.F90
+++ b/src/SIS_slow_thermo.F90
@@ -1498,7 +1498,7 @@ subroutine SIS_slow_thermo_init(Time, G, US, IG, param_file, diag, CS, tracer_fl
 
   call get_param(param_file, mdl, "TRANSMUTE_SEA_ICE", CS%transmute_ice, &
                  "If true, allow ice to be transmuted directly into seawater with a spatially "//&
-                 "spatially varying rate as a form of outflow open boundary condition.", &
+                 "varying rate as a form of outflow open boundary condition.", &
                  default=.false.)
   if (CS%transmute_ice) then
 
diff --git a/src/ice_model.F90 b/src/ice_model.F90
index ebf320c..da0b7e4 100644
--- a/src/ice_model.F90
+++ b/src/ice_model.F90
@@ -1956,7 +1956,7 @@ subroutine ice_model_init(Ice, Time_Init, Time, Time_step_fast, Time_step_slow,
                  default=.false., do_not_log=.true.) ! Defer logging to SIS_slow_thermo.
   call get_param(param_file, mdl, "TRANSMUTE_SEA_ICE", transmute_ice, &
                  "If true, allow ice to be transmuted directly into seawater with a spatially "//&
-                 "spatially varying rate as a form of outflow open boundary condition.", &
+                 "varying rate as a form of outflow open boundary condition.", &
                  default=.false., do_not_log=.true.) ! Defer logging to SIS_slow_thermo.
 
 
```

**Comment by Hallberg-NOAA** [comment]
Thanks for the review and the suggestion, @kshedstrom. The duplicate words have been corrected as suggested.

**Comment by marshallward** [comment]
I've run this through my "Python regression" and all runs have reproduced, so I will merge this now.

An additional parameter update will be required for MOM6-examples, which I'll also push shortly.
